packages/ag-charts-community/src/chart/agChartV2.ts

```diff
@@ -425,20 +430,36 @@ function applyModules(chart: Chart, options: AgChartOptions) {
     };
 
     let modulesChanged = false;
-    const rootModules = REGISTERED_MODULES.filter((m): m is RootModule => m.type === 'root');
-    for (const next of rootModules) {
-        const shouldBeEnabled = matchingChartType(next) && (options as any)[next.optionsKey] != null;
-        const isEnabled = chart.isModuleEnabled(next);
-
-        if (shouldBeEnabled === isEnabled) continue;
-        modulesChanged = true;
-
-        if (shouldBeEnabled) {
-            chart.addModule(next);
-        } else {
-            chart.removeModule(next);
+    const processModules = <T extends Module<ModuleInstance>>(
+        moduleType: string,
```

Comment:
Is there a stricter type we can use here rather than `string`? Would `Module['type']` work?